app/internal/languages.py

```diff
@@ -34,14 +32,12 @@ def set_ui_language(language: str = None) -> None:
     # if not language:
     #     language = _get_display_language(user_id: int)
 
-    language_dir = _get_language_directory()
-
-    if language not in list(_get_supported_languages(language_dir)):
+    if language not in list(_get_supported_languages()):
```

Comment:
Maybe `set(_get_supported_languages)` to get faster results?

You can also omit the casting to `list` if the performance is not an issue

Reply:
Probably misunderstood you here, so would appreciate further clarification:
```
>       if language not in set(_get_supported_languages()):
E       TypeError: unhashable type: 'list
```

Reply:
fixed.